File: binary_search/부품찾기.py
# 부품찾기

import sys

input = sys.stdin.readline
n = int(input())
nums = list(map(int, input().split()))
m = int(input())
checks = list(map(int, input().split()))

# 이진 탐색을 위해서는 반드시 정렬 필요
nums.sort()

def binary_search(li, t, s, e):
  while s <= e:
    mid = (s + e) // 2
    if li[mid] == t:
      return 'yes'
    elif li[mid] < t:
      s = mid + 1
    else:
      e = mid - 1
  return 'no'

for c in checks:
  print(binary_search(nums, c, 0, len(nums)-1), end=' ')

File: 이것이취업을위한코딩테스트다/이진탐색/부품 찾기.js
const fs = require('fs');
const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const N = Number(inputs[0]);
const tools = inputs[1].split(' ').map(Number);
const M = Number(inputs[2]);
const orders = inputs[3].split(' ').map(Number);
// console.log(N, tools, M, orders)

tools.sort((a, b) => a - b);
// console.log(tools)

function binarySearch(arr, t, s, e) {
  while (s <= e) {
    const m = Math.floor((s + e) / 2);
    if (arr[m] === t) return 'yes';
    else if (arr[m] > t) e = m - 1;
    else s = m + 1;
  }
  return 'no';
}

const result = orders.map((item) => binarySearch(tools, item, 0, N - 1));
// console.log(result)

console.log(result.join(' '));

// 집합 자료형 기반 풀이

const fs2 = require('fs');
const inputs2 = fs2.readFileSync('/dev/stdin').toString().trim().split('\n');

const N2 = Number(inputs2[0]);
const items = new Set(inputs2[1].split(' ').map(Number));
const M2 = Number(inputs2[2]);
const orders2 = inputs2[3].split(' ').map(Number);
// console.log(N2, items, M2, orders2)

const result2 = orders2.map((item) => {
  if (items.has(item)) return 'yes';
  else return 'no';
});

console.log(result2.join(' '));
